src/MagnumPlugins/UfbxImporter/Test/UfbxImporterTest.cpp

```diff
@@ -357,7 +843,1651 @@ void UfbxImporterTest::light() {
+        Containers::StridedArrayView1D<const Vector3> position = mesh->attribute<Vector3>(MeshAttribute::Position);
+        Containers::StridedArrayView1D<const Vector3> normal = mesh->attribute<Vector3>(MeshAttribute::Normal);
+        for(UnsignedInt i = 0; i < vertexCount; ++i) {
+            CORRADE_ITERATION(i);
+            CORRADE_COMPARE(position[i].z(), -1.0f);
+            CORRADE_COMPARE(normal[i], (Vector3{0.0f,0.0f,-1.0f}));
+        }
+    }
+
+}
+
+void UfbxImporterTest::instancedMaterial() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("UfbxImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Path::join(UFBXIMPORTER_TEST_DIR, "instanced-material.fbx")));
+    CORRADE_VERIFY(importer->isOpened());
+    CORRADE_COMPARE(importer->sceneCount(), 1);
+    CORRADE_COMPARE(importer->objectCount(), 3);
+    CORRADE_COMPARE(importer->materialCount(), 6);
+    CORRADE_COMPARE(importer->meshCount(), 4);
+
+    Long modelA = importer->objectForName("ModelA");
+    Long modelB = importer->objectForName("ModelB");
+    Long modelC = importer->objectForName("ModelC");
+    Int materialA = importer->materialForName("MaterialA");
+    Int materialB = importer->materialForName("MaterialB");
+    Int materialC = importer->materialForName("MaterialC");
+    Int topA = importer->materialForName("TopA");
+    Int topB = importer->materialForName("TopB");
+    Int topC = importer->materialForName("TopC");
+
+    CORRADE_COMPARE_AS(modelA, 0, TestSuite::Compare::GreaterOrEqual);
+    CORRADE_COMPARE_AS(modelB, 0, TestSuite::Compare::GreaterOrEqual);
+    CORRADE_COMPARE_AS(modelC, 0, TestSuite::Compare::GreaterOrEqual);
+    CORRADE_COMPARE_AS(materialA, 0, TestSuite::Compare::GreaterOrEqual);
+    CORRADE_COMPARE_AS(materialB, 0, TestSuite::Compare::GreaterOrEqual);
+    CORRADE_COMPARE_AS(materialC, 0, TestSuite::Compare::GreaterOrEqual);
+    CORRADE_COMPARE_AS(topA, 0, TestSuite::Compare::GreaterOrEqual);
+    CORRADE_COMPARE_AS(topB, 0, TestSuite::Compare::GreaterOrEqual);
+    CORRADE_COMPARE_AS(topC, 0, TestSuite::Compare::GreaterOrEqual);
+
+    Containers::Optional<SceneData> scene = importer->scene(0);
+    CORRADE_VERIFY(scene);
+
+    CORRADE_COMPARE_AS(scene->meshesMaterialsFor(UnsignedLong(modelA)), (Containers::arrayView<Containers::Pair<UnsignedInt, Int>>({
+        { 0, materialA }, { 1, topA },
+    })), TestSuite::Compare::Container);
+
+    CORRADE_COMPARE_AS(scene->meshesMaterialsFor(UnsignedLong(modelB)), (Containers::arrayView<Containers::Pair<UnsignedInt, Int>>({
+        { 2, materialB }, { 3, topB },
+    })), TestSuite::Compare::Container);
+
+    /* Note: Re-uses mesh from A with different materials! */
+    CORRADE_COMPARE_AS(scene->meshesMaterialsFor(UnsignedLong(modelC)), (Containers::arrayView<Containers::Pair<UnsignedInt, Int>>({
+        { 0, materialC }, { 1, topC },
+    })), TestSuite::Compare::Container);
+}
+
+void UfbxImporterTest::instancedMaterialDisabled() {
```

Comment:
Yeah it's a weird setting, no objections for removing it. Initially I had it because I was unsure if it would break user code assuming one mesh maps to one material, for example the Blender FBX importer fails to load instanced meshes with multiple materials (actually, it won't even export it so I had to poke it a bit to make that test case file :D). Other programs do export files like that though so it's not a purely theoretical feature.

Reply:
In case of Magnum, user code that expects each mesh to only ever have one material would have to deal with that for any format, not just FBX, so that's something that should go into a format-independent processing utility instead of being a plugin-specific config option.